src/ex5/server/services/item_manager.js

```diff
@@ -0,0 +1,93 @@
+// The ItemManager should go here. Remember that you have to export it.
+
+const { Item } = require('../db/models');
+
+const SORT_OPTIONS = ['id', 'text', 'status']
+
+class ItemManager {
+  init() {
+    this.sortOptionIndex = 0;
+    this.sortBy = SORT_OPTIONS[this.sortOptionIndex];
+  }
+
+  async addItem(text) {
+    try {
+      return await Item.create({
+        text: text,
+        isNew: true,
+        status: false,
+        done: null
+      });
+    } catch (error) {
+      return await Item.update({
+        isNew: true,
+      }, {
+        where: { text: text }
+      })
+    }
+  }
+
+  async getItem(itemId) {
+    return await Item.findAll({
+      where: {
+        id: itemId
+      }
+    });
+  }
+
+  async updateItem(itemId, body) {
+    return await Item.update({
+      text: body.text,
+      isNew: body.isNew,
+      status: body.status,
+      done: body.done
+     }, {
+      where: { id: itemId }
+     })
+  }
+
+  async markTodoAsOld(itemId) {
+    return await Item.update({
+      isNew: false,
+    }, {
+      where: { id: itemId }
+    })
+  }
+
+  async changeTodoStatus(itemId, body) {
+    return await Item.update({
+      status: body.status,
+      done: body.done
+    }, {
+      where: { id: itemId }
+    })
+  }
+
+  async deleteItem(itemId) {
+    return await Item.destroy({
+      where: { id: itemId },
+    });
+  }
+
+  async clearAll() {
+    return await Item.destroy({
```

Comment:
Might be better to mark them as deleted in some way (i.e. status column) instead of destroying everything, there's no way back from that :)
I know potentially in the future you will only be destorying those that correspond to some id (account, user, etc.) but still at the moment this is a scary function :)

Reply:
I like this clearAll because it really destroys all the data and new ids are calculated from the beginning again. In ex6 I added "are you sure"-confirmation.